sktime/forecasting/compose/_multiplexer.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+from sktime.forecasting.base._meta import _HeterogenousEnsembleForecaster
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+from sklearn.base import clone
+
+__author__ = ["Kutay Koralturk"]
+__all__ = ["MultiplexForecaster"]
+
+
+class MultiplexForecaster(
+    _OptionalForecastingHorizonMixin, _HeterogenousEnsembleForecaster
+):
+    def __init__(
+        self,
+        forecasters: list,
+        selected_forecaster=None,
+    ):
+
+        """
+        MultiplexForecaster facilitates a framework for performing
+        model selection process over different model classes.
+        It should be used in conjunction with ForecastingGridSearchCV
+        to get full utilization.
+
+        Single use of MultiplexForecaster with forecasters
+        and selected_forecaster parameter specified,
+        works just like the selected component.
+        It does not provide any further use in that case.
+
+        When used with ForecastingGridSearchCV, MultiplexForecaster
+        provides an ability to compare different model class
+        performances with each other, just like a model tournament.
+        When ForecastingGridSearchCV is fitted with a MultiplexForecaster,
+        returned value for the selected_forecaster argument of best_params_
+        attribute of ForecastingGridSearchCV, gives the best
+        performing model class among given models provided in forecasters.
+
+        Parameters
+        ----------
+        forecasters : list
+            List of (forecaster names, forecaster objects)
+            MultiplexForecaster switches between these forecasters
+            objects when used with ForecastingGridSearchCV to
+            find the optimal model
+        selected_forecaster: str
+            An argument to make a selection among forecasters.
+            MultiplexForecaster uses selected_forecaster
+            to choose which component to fit.
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that MultiplexForecaster will currently
+            forecast with.
+        """
+
+        super(MultiplexForecaster, self).__init__(forecasters=forecasters, n_jobs=None)
+        self.selected_forecaster = selected_forecaster
+        self._forecaster = None
+        self._check_forecasters()
```

Comment:
That should be called at the beginning of `fit` rather than `__init__`, the reason is that one may change the forecasters later through `set_params` without calling `__init__` again.

Reply:
I agree